client/my-sites/themes/theme-showcase.jsx

```diff
@@ -118,17 +118,21 @@ const ThemeShowcase = React.createClass( {
 		trackClick( 'upload theme' );
 	},
 
+	showUploadButton() {
+		const { siteSlug, isMultisite, isJetpack } = this.props;
+
+		return config.isEnabled( 'manage/themes/upload' ) && siteSlug && ! isMultisite &&
+			( isJetpack || config.isEnabled( 'automated-transfer' ) );
```

Comment:
It's taken me a while to get used to formatting long boolean clauses like this in an extra set of `()` like this:

```js
return(
    config.isEnabled( 'manage/themes/upload' ) &&
    siteSlug &&
    ! isMultisite &&
    ( isJetpack || config.isEnabled( 'automated-transfer' ) )
);
```